text/0101-new-scale-control-model-based-on-IO-size.md

```diff
@@ -0,0 +1,82 @@
+
+### How to adjust windows size 
+The minimum(initial) value of the sliding window is 1000MB. The store can be considered to be scheduled if it's available sending size is bigger than 100MB, so the biggest region can also be scheduled. 
+When the operator execution duration is less than 2 times the snapshot task execution time, the sender windows size will increase , otherwise it will decrease. A PI controller is used to prevent repeated adjustments. It should be noted that when all schedules don't need such big size, it will stop increasing. 
+
+### How to ensure fairness 
+The new scheduler policy does not guarantee that the snapshot sending size of all stores is exactly the same, but will switch to the next candidate region when the store has no available sending size. During the prototype testing process on the scale-out and scale-in scene , the sending size on all TIKV reaches the bottleneck. 
+
+### Scheduler 
+#### Balance Region Scheduler 
+The scheduler picks the store as source first, then picks the region on the source store, finally picks the target store. On the original basis, it is necessary to check the target region leader whether there is enough sending space to send. 
+#### High Priority Scheduler 
+Some scheduler may create a highest priority operator such as balance hot region, region check. For these higher operators, 20% of the send window will be reserved. This can prevent this window size from being completely occupied by low priority task .
+
+## Compatibility
+### Multi RocksDB
+In this design, the generation of the snapshot will use check-point hard link to replace the scanning of RocksDB. It can reduce the CPU loads and the duration of snapshot generator. But the snapshots are not compressed ,the network and disk bandwidth will increase. 
+The snapshot synchronization process is basically the same, some tasks need to wait to be executed when the store has many snapshot tasks. The waiting time is still positively related to the backlog of the snapshot task, so feedback mechanism still satisfies this situation. 
+## Alternatives
+### Follower Replica 
+In this design, a new mechanism in raft algorithm will be introduced which allows the follower to replaica log or send snapshot to another peer. Since the removed peer can also be the snapshot sender, balance region scheduler no need to consult the region leader whether there is some available sending size. 
+But for the other scheduler, there may be a big log gap between the follower and leader, so the follower cannot be the snapshot sender, this may make the operator slower and complicated.
+Drawbacks
+Due to the need to prevent too many snapshot tasks on the instance, a new limit needs to be added, which may cause the execution priority of the scheduler to change, resulting in a small number of scheduling. 
+
+## Question
+1. How about removing the store limit?
```

Comment:
How about adding a section to clarify the relationship between this and the store limit?

Reply:
move to compatibility